pkg/eventconsumer/event_consumer.go

```
@@ -477,3 +616,14 @@ func (ec *eventConsumer) Close() error {
 
 	return nil
 }
+
+func sessionTypeFromKeyType(keyType types.KeyType) mpc.SessionType {
+	switch keyType {
+	case types.KeyTypeSecp256k1:
+		return mpc.SessionTypeECDSA
+	case types.KeyTypeEd25519:
+		return mpc.SessionTypeEDDSA
+	default:
+		panic(fmt.Sprintf("unsupported key type: %v", keyType))
```

avoid panic in production as it makes the server crash and unrecoverable, should return an error instead.